students/Amit/Session02/series.py

```diff
@@ -0,0 +1,22 @@
+#! /usr/bin/env python
+
+fib_cache= {}
+
+def fibonacci(n):
+    if n in fib_cache:
+        return fib_cache[n]
+
+    if n == 0:
+        return 0
+    elif n == 1:
+        return 1
+    elif n == 2:
```

Comment:
no need for this check -- the algorithm will work for n == 2 -- and you'll be caching the result anyway.